src/typed_document.cpp

```diff
@@ -0,0 +1,58 @@
+#include <typed_document.hpp>
+#include <dao.hpp>
+
+namespace hypha
+{
+    template<typename std::string& T>
+    TypedDocument<T>::TypedDocument(dao& dao, const eosio::checksum256& hash)
+    : Document(dao.get_self(), hash), m_dao(dao)
+    {
+        this->validate();
```

Comment:
I'd suggest to omit `this->` to align with the project style.

Reply:
:cry: